drivers/azure/azure.go

```diff
@@ -415,7 +426,7 @@ func (d *Driver) Create() error {
 	if err := c.CreateAvailabilitySetIfNotExists(ctx, d.deploymentCtx, d.ResourceGroup, d.AvailabilitySet, d.Location, d.ManagedDisks, int32(d.FaultCount), int32(d.UpdateCount)); err != nil {
 		return err
 	}
-	if err := c.CreateNetworkSecurityGroup(ctx, d.deploymentCtx, d.ResourceGroup, d.naming().NSG(), d.Location, d.deploymentCtx.FirewallRules); err != nil {
+	if err := c.ConfigureNetworkSecurityGroup(ctx, d.deploymentCtx, d.nsgResource, d.ResourceGroup, d.Location, d.nsgUsedInPool, d.BaseDriver.SSHPort, d.DockerPort, d.OpenPorts); err != nil {
```

Comment:
Why not just pass in d? That would clean up this func signature a lot.

Reply:
Typically the `azureutil` package doesn't reference the `azure` package at all as we provide the expected values necessary from the driver when making the calls, i.e.:

https://github.com/rancher/machine/blob/47ba74901e3084288fc72826b68fdb311bbb810d/drivers/azure/azure.go#L448-L450

Passing in `d` is definitely an option, but this would be a break from the way we normally make calls using the `azureutil` package.